Breadcrumbs.py

```diff
@@ -32,80 +38,169 @@ def get_row_indentation(points, view, tab_size, limit=1e20):
 def is_white_row(view, points):
   return all(view.substr(pt).isspace() for pt in reversed(points))
 
-def get_breadcrumb(view, points, limit):
+def get_breadcrumb(view, points, regex, separator):
   for pt in points:
     ch = view.substr(pt)
     if not ch.isspace():
-      return view.substr(sublime.Region(pt, min(view.line(pt).b, pt + limit))).strip()
+      linestring = view.substr(sublime.Region(pt, min(view.line(pt).b, pt + 500))).strip()
```

Comment:
I kind of liked the way strip presents a cleaner string for the regex to work with. But you're right, letting the regex do everything makes it simpler and more flexible.